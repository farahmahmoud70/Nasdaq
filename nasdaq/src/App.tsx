import './App.css';
import { Layout } from './app/components/layout/Layout';
import Header from './app/components/header/Header';
import Footer from './app/components/footer/Footer';
import { GlobalStyles } from './app/style/Global';
import AppRoutes from './Routes';

const App = () => {

  return <>
    <GlobalStyles />
    <Layout>
      <Header />
      <AppRoutes />
      <Footer />
    </Layout>
  </>
}

export default App;
