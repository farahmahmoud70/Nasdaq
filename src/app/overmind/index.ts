import { IContext } from 'overmind';
import {
  createStateHook,
  createActionsHook,
  createEffectsHook,
  createReactionHook,
} from 'overmind-react';
import { namespaced } from 'overmind/config';
import * as stocks from 'app/overmind/stocks/stocksIndex';
import * as stockDetails from 'app/overmind/stockDetails/stockDetailsIndex';

export const config = namespaced({
  stocks,
  stockDetails,
});

export const devTool = {
  devtools: true, // defaults to 'localhost:3031'
};

export type Context = IContext<typeof config>;

export const useAppState = createStateHook<Context>();
export const useActions = createActionsHook<Context>();
export const useEffects = createEffectsHook<Context>();
export const useReaction = createReactionHook<Context>();
