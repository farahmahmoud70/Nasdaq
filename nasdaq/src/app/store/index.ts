
import { IContext } from 'overmind'
import {
    createStateHook,
    createActionsHook,
    createEffectsHook,
    createReactionHook
} from 'overmind-react'
import { namespaced } from 'overmind/config'
import * as stocks from './stocks/stocksIndex'
// import * as stockDetails from './stockDetails/stockDetailsIndex'

export const config = namespaced({
    stocks,
    // stockDetails
})


export type Context = IContext<typeof config>

export const useAppState = createStateHook<Context>()
export const useActions = createActionsHook<Context>()
export const useEffects = createEffectsHook<Context>()
export const useReaction = createReactionHook<Context>()