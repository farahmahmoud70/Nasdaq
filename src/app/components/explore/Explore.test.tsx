import { cleanup, screen, waitFor } from '@testing-library/react';
import App from 'App';
import { config } from 'app/overmind';
import { createOvermindMock } from 'overmind';
import { Provider } from 'overmind-react';
import { BrowserRouter } from 'react-router-dom';

afterEach(cleanup);

beforeEach(() => {
  const overmind = createOvermindMock(config);
  <Provider value={overmind}>
    <BrowserRouter>
      <App />
    </BrowserRouter>
  </Provider>;
});

describe('Explore page', () => {
  it('shows correct snapshot for the explore wrapper styled component', () => {
    const exploreWrapperElm = screen.findByTestId('explore-page');
    expect(exploreWrapperElm).toMatchSnapshot();
  });

  it('shows correct snapshot for the explore title styled component', async () => {
    const exploreTitleElm = screen.findByTestId('explore-title');
    waitFor(() => expect(exploreTitleElm).toHaveStyleRule('width', '100%'));
    expect(exploreTitleElm).toMatchSnapshot();
  });

  it('shows correct loading state before content loads', async () => {
    const exploreLoaderElm = screen.findByTestId('loader');
    waitFor(() => {
      expect(exploreLoaderElm).toBeInTheDocument();
    });
  });
});
