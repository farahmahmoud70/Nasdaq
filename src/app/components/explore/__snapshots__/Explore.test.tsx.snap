// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Explore page shows correct snapshot for the explore title styled component 1`] = `Promise {}`;

exports[`Explore page shows correct snapshot for the explore wrapper styled component 1`] = `Promise {}`;
